Add use case guide for HTTP routing with Traefik (#21023)
docker/docs: content/guides/use-case/traefik.md

In this guide, you will learn how to:

1. Start Traefik with Docker
2. Configure routing rules to split traffic between two containers
3. Use Traefik in a containerized development environment
4. Use Traefik to send requests to non-containerized workloads

## Prerequisites

The following prerequisites are required to follow along with this how-to guide:

- [Docker Desktop](https://www.docker.com/products/docker-desktop/)
- [Node.js](https://nodejs.org/en/download/package-manager) and [yarn](https://yarnpkg.com/)
- Basic of Docker

## Using Traefik with Docker

One of the unique features of Traefik is its ability to be configured in many ways. When using the Docker provider, Traefik gets its configuration from other running containers using [labels](https://docs.docker.com/config/labels-custom-metadata/). Traefik will watch engine events (for container starts and stops), extract the labels, and update its configuration.

> The two most common Traefik-monitored labels will be ([view the full list here](https://doc.traefik.io/traefik/routing/providers/docker/)):

Reviewer: Not a blocker, just a small thing I noticed. the double ()s cause an extra space in the deploy preview 
![image](https://github.com/user-attachments/assets/d460779e-b2c4-4191-af58-bd8f4a4b226a)

Author: Huh! Any idea how to resolve that seeing I have no space in the text itself? Could always just rewrite that segment to take it out of the ()s too.

Maybe something like this?

```suggestion
While there are [many Traefik-monitored labels](https://doc.traefik.io/traefik/routing/providers/docker/), the two most common will be:
```

Reviewer: I like that fix!

    ```

3. Now, start a simple Nginx container and define the labels Traefik is watching for to configure the HTTP routing. Note that the Nginx container is not exposing any ports.

    ```console
    $ docker run -d --network=traefik-demo --label 'traefik.http.routers.nginx.rule=Host(`nginx.localhost`)' nginx
    ```

    Once the container starts, open your browser to [http://nginx.localhost](http://nginx.localhost) to see the app (all Chromium-based browsers route *.localhost requests locally with no additional setup).

4. Start a second application that will use a different hostname.

    ```console
    $ docker run -d --network=traefik-demo --label 'traefik.http.routers.welcome.rule=Host(`welcome.localhost`)' docker/welcome-to-docker
    ```

    Once the container starts, open your browser to http://welcome.localhost. You should see a “Welcome to Docker” website.

## Using Traefik in development

Now that you’ve experienced Traefik, it’s time to try using it in a development environment. In this example, you will use a sample application that has a split frontend and backend. This app stack has the following configuration:

> 1. All requests to /api to go to the API service

Reviewer: should /api be in code? `/api` 
same with Step 3: `db.localhost`
